contracts/SL2RD.sol

```diff
@@ -62,21 +83,47 @@ contract SL2RD is
     /// owner for each tokenId. This is prefilled with the owner's
     /// address for each token, but designed with flexibilty to include
     /// others during initialization if they are known.
+    /// The communitySplitsBasisPoints is a const percentage denoted in
+    /// basis points for how many of the slots the creator initially wants
+    /// to allocate for everyone else.
+    /// @param addresses_ The addresses for initial distribution.
+    /// @param tokenIds_ The tokenIds for initial distribution.
+    /// @param communitySplitsBasisPoints_ The percentage of slots allocated
+    /// for the community (denoted in basis points).
+    /// @param shareContractAddress_ The address of the share contract.
+    /// @param operatorRegistryAddress_ The address of the share operator registry.
     function initialize(
         address[] memory addresses_,
         uint256[] memory tokenIds_,
-        address shareContractAddress_
+        uint256 communitySplitsBasisPoints_,
+        address shareContractAddress_,
+        address operatorRegistryAddress_
     ) public onlyOwner {
         require(tokenIds_.length == addresses_.length, "SHARE028");
         require(tokenIds_.length <= MAX_SPLIT_COUNT, "SHARE006");
         require(tokenIds_.length >= 1, "SHARE020");
+        require(communitySplitsBasisPoints_ <= MAX_BASIS_POINTS, "SHARE029");
+        // check operatorAddress
+
         setShareContractAddress(shareContractAddress_);
         _protocol = SHARE(shareContractAddress_);
 
+        _shareOperatorRegistry = OperatorRegistry(operatorRegistryAddress_);
+        Immutable.setUnsignedInt256(_totalSlots, tokenIds_.length);
+        Immutable.setUnsignedInt256(
+            _communitySplitsBasisPoints,
+            communitySplitsBasisPoints_
+        );
+        Immutable.setAddress(_initialOwner, addresses_[0]);
+
+        _totalCommunitySlots =
```

Comment:
Is this safe?

Reply:
@un7c0rn  yes. 

Despite us using predefined ranges (e.g 10, 100, 200, 250, 1000), the basis points would still produce decimal values for ownership allocations since the basis points allow for precision. We could only get exact numbers of addresses if we also curate the percentages for the community/private allocations. BUT none of this really matters because when dividing by the max basis points (ie 10000) the decimal is truncated (e.g 86.56% aka 8656 bp is the community allocation of my 250 address => 216.4, then in solidity this truncates to 216 ) simply, all will still be under the initial owner address and you will just distribute 216 since a slot cannot be fractionalized in its current state (NFT).

Where a problem is introduced is when you have an unusual commuynity split percentage, as above, and then want to also either give each participant a certain % or break up the community splits by %. So if we take the same numbers as above and now want to divide the splits by 5% chunks we would not be able to properly. However, this is something we can simply handle via the UX with curated configurations and patterns for splits.

Reply:
+1! Can you add this commentary to your doc?